packages/provider-elasticsearch/src/example-types/results/highlighting/request.js

```diff
@@ -0,0 +1,100 @@
+import _ from 'lodash/fp.js'
+import F from 'futil'
+import { CartesianProduct } from 'js-combinatorics'
+
+/**
+ * Return names for all fields groups and their sub-fields in `schema`.
+ */
+const getFieldsGroupsNames = _.memoize((schema) => {
+  const subFields = _.keys(
+    _.pickBy('highlight', schema.elasticsearch?.subFields)
+  )
+
+  const fields = _.flatMap((field) => {
+    const copy_to = field.elasticsearch?.copy_to
+    if (_.isEmpty(copy_to)) return copy_to
+    const product = new CartesianProduct(copy_to, subFields)
+    return [...copy_to, ...Array.from(product).map(_.join('.'))]
+  }, schema.fields)
+
+  return new Set(fields)
+}, _.get('elasticsearch.index'))
+
+/**
+ * Return mappings for all sub-fields that can be highlighted in `mapping`.
+ */
+const getSubFieldsMappings = (schema, mapping) =>
+  _.flow(
+    F.pickByIndexed((v, k) => schema.elasticsearch?.subFields?.[k]?.highlight),
+    F.mapValuesIndexed((v, k) => ({
+      ...v,
+      meta: mapping.meta,
+      copy_to: _.map((multi) => `${multi}.${k}`, mapping.copy_to),
+    }))
+  )(mapping.fields)
+
+/**
+ * Return mappings for all fields and their sub-fields that can be highlighted
+ * in `schema`.
+ */
+const getFieldsMappings = _.memoize((schema) => {
+  const fieldsGroups = getFieldsGroupsNames(schema)
+  return F.reduceIndexed(
+    (acc, { elasticsearch: mapping }, name) => {
+      // Only include leaf fields (have mapping) and do not include fields
+      // groups.
+      if (mapping && !fieldsGroups.has(name)) {
+        Object.assign(acc, {
+          [name]: mapping,
+          ..._.mapKeys(
+            (k) => `${name}.${k}`,
+            getSubFieldsMappings(schema, mapping)
+          ),
+        })
+      }
+      return acc
+    },
+    {},
+    schema.fields
+  )
+}, _.get('elasticsearch.index'))
+
+export const getRequestBodyHighlight = (schema, node, config) => {
+  const query = node._meta?.relevantFilters
+  const querystr = JSON.stringify(query)
+  const allFieldsGroups = getFieldsGroupsNames(schema)
+
+  // Pre-computed list of fields groups present in the query
+  const queryFieldsGroups = []
+  F.walk()((val, key) => {
+    if (allFieldsGroups.has(val)) queryFieldsGroups.push(val)
+    if (allFieldsGroups.has(key)) queryFieldsGroups.push(key)
+  })(query)
+
+  const getHighlightQuery = (mapping, name) => {
+    const toReplace = _.intersection(queryFieldsGroups, mapping.copy_to)
+    if (!_.isEmpty(toReplace)) {
+      const regexp = new RegExp(_.join('|', toReplace), 'g')
+      return JSON.parse(_.replace(regexp, name, querystr))
+    }
+  }
+
+  const mappingToHighlightConfig = (mapping, name) => {
+    const isBlob = mapping.meta?.subType === 'blob'
```

Comment:
Maybe we can put the fragment size and number of fragments in the meta data here so that it is configurable based on the mappings with the values here as sane defaults still?

Reply:
Elastic's `meta` has lots of restrictions https://www.elastic.co/guide/en/elasticsearch/reference/8.11/mapping-field-meta.html so we can't just dump configuration there. The reason we're using `subType` at all (instead of specifying the subType in the schema overrides) is to leverage syncing to set the subType if possible.

I think your ask here is to have an API for allowing highlight configuration which we're trying to avoid with this PR. Not that we will never have it but we're trying to get away with no configuration whatsoever if we can for now.

Reply:
I am thinking we could just add fragmentCount and also fragmentSize to meta, I get that we are limited to 5 based on the documentation, but this just seems more flexible in general to rely on the mappings for these fields instead of the schema. I understand though if we do not want to based on the limitations of the meta field.

Reply:
Yeah `subType` is ok because it's intrinsic to the field and it's a small payload. If we're going to allow highlighting configuration there should ideally be one way of doing it and `meta` cannot accommodate for arbitrary configuration.

Reply:
We decided to set these flags in the field schema instead of elastic's `meta`